### ISSUE TITLE
Draw calls in 2d should be drawn on Z = 0, or on top of 3d content?

### ISSUE THREAD

**Issue by julioobelleiro** [CingProject/Cing#9]
Right now a call like image.draw(x, y) does draw the image in screen coordinates being on top of any other 3d content (unless the render queue for that object is changed from the default).

It might be more intuitive that draw(x,y) just draws the object with Z = 0, but in 3d space?

**Comment by f0ster** [comment]
If someone plans on both objects in 3d space, an image for example, and then also plans on drawing a 2d object, would the user expect it to always be closest to the screen ? or will they want to position 2d drawings at some depth with a z-coordinate?  I think it makes more sense to have the z-planes relative to the same origin (0) instead of whatever the screen-most layer is.

**Comment by julioobelleiro** [comment]
Yes, that seems more intuitive. One possibility we have discussed to fix this would be:
1. draw calls with just x, y, render the object (whatever it is) in the plane z = 0
2. we add calls called drawBackground and drawForeground to allow to draw objects on the back most and on the front most. This would be an easy way to have some control over this, but still be super simple.
3. for those who need more control, we can add the renderqueue(int) GraphicsUserAPI function to set the render queue for the upcoming draw calls. (kind of like the translate or rotate calls work, setting the state for the following calls until it is changed).
4. finally, the remaining problem is to have a better handling on the camera far/near plane range, so that different z values affect the render order, even if they are small (like 0.1 and 0.2 would be drawn at different distances from the camera)
